### DISCUSSION TITLE
Controller as dependency of another controller

### DISCUSSION THREAD

**Discussion by diego-lipinski-de-castro** [jonataslaw/getx#1336] [Q&A]
Is it ok to have a controller to be injected into another controller? I am in this problem where I have a controller (eg a global controller, or home controller), and I need to use it inside other controllers, is this ok or i am probably doing something wrong?

**Comment by ad-on-is** [comment]
Depends what you mean by controller. If your FooWidget uses GetBuilder with FooController, then you MUST NOT use FooController in another Controller. 

What you can do instead, is Get.put(SomeService()) and Get.find<SomeService>() wherever you want. See GetService in the docs

**Reply by diego-lipinski-de-castro** [reply]
Yeah that's what I ended up doing, to use a service seems the right way to go
